DatadogSDKBridge/Classes/Implementation/AnyEncodable.swift

```diff
@@ -6,16 +6,53 @@
 
 import Foundation
 
-internal func castAttributesToSwift(_ attributes: NSDictionary) -> [String: AnyEncodable] {
+internal func castAttributesToSwift(_ attributes: NSDictionary) -> [String: Encodable] {
     return castAttributesToSwift(attributes as? [String: Any] ?? [:])
 }
 
-internal func castAttributesToSwift(_ attributes: [String: Any]) -> [String: AnyEncodable] {
-    return attributes.mapValues { AnyEncodable($0) }
+internal func castAttributesToSwift(_ attributes: [String: Any]) -> [String: Encodable] {
+    var casted: [String: Encodable] = [:]
+
+    attributes.forEach { key, value in
+        if key.hasPrefix("_dd.") {
```

Comment:
Instead of relying to a specific prefix, can't we use this trick for all value ? If it's a known type, use the casted value (string, number, etc...), otherwise use the AnyEncodable ?
Something like: 
```swift
switch value {
    case let string as String: // unpacking `NSTaggedPointerString`
        return string
    case let int64 as Int64: // unpacking `__NSCFNumber`
        return int64
    case let double as Double: // unpacking `__NSCFNumber` again; trying `Double` after `Int` as it's a wider type
        return double
    // … handle a few more numbers and known types
    default:
        return AnyEncodable(value)
}
```

Reply:
It's a very good idea 👍. I updated this PR with that implementation, but I made it more explicit to differentiate that we first try to preserve type information but if not possible we fallback to preserving encoded value (data representation):
```swift
attributes.forEach { key, value in
    if let castedValue = castByPreservingTypeInformation(attributeValue: value) {
        // If possible, cast attribute by preserving its type information
        casted[key] = castedValue
    } else {
        // Otherwise, cast by preserving its encoded value (and loosing type information)
        casted[key] = castByPreservingEncodedValue(attributeValue: value)
    }
}
```